File: ScreenshotPreview/ScreenshotPreview/NSString+CommandLineScript.m
//
//  NSString+CommandLineScript.m
//  RunScriptTest
//
//  Created by Jack Rostron on 02/05/2014.
//  Copyright (c) 2014 Jack Rostron. All rights reserved.
//

#import "NSString+CommandLineScript.h"

@implementation NSString (CommandLineScript)

- (NSString *)commandLineOutput
{
    NSTask *task = [[NSTask alloc] init];
    NSPipe *pipe = [NSPipe pipe];
    NSFileHandle *file = [pipe fileHandleForReading];
    
    [task setLaunchPath: @"/bin/sh"];
    [task setArguments:@[@"-c", self]];
    [task setStandardOutput: pipe];
    [task launch];
    
    return [[NSString alloc] initWithData:[file readDataToEndOfFile] encoding:NSUTF8StringEncoding];
}

@end

File: Swift/iOS-UIAutomator/iOS-UIAutomator/NSString+CommandLineScript.swift
//
//  NSString+CommandLineScript.swift
//  SwiftCommandLine
//
//  Created by Jack Rostron on 16/06/2014.
//  Copyright (c) 2014 Jack Rostron. All rights reserved.
//

import Cocoa

extension String {
    
    func commandLineOutput() -> String {
        var task = NSTask()
        var pipe = NSPipe.pipe()
        var fileHandle = pipe.fileHandleForReading
        
        task.launchPath = "/bin/sh"
        task.arguments = ["-c", self]
        task.standardOutput = pipe
        task.launch()
        
        var data = fileHandle.readDataToEndOfFile()
        return NSString(data: data, encoding: NSUTF8StringEncoding)
    }
    
}